Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -5352,6 +5408,8 @@ template <class TreeBuilder> TreeExpression Parser<LexerType>::parseMemberExpres
     } else if (!baseIsNewTarget) {
         const bool isAsync = matchContextualKeyword(m_vm.propertyNames->async);
 
+        if (TreeExpression argumentsDotLengthExpression = tryParseArgumentsDotLengthForFastPath(context, startLocation, expressionStart))
+            return argumentsDotLengthExpression;
```

Comment:
Why do we return here? Why not just return ResolveNode with `arguments` without marking `usesArguments` and use the rest of the code as is?